src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -50,42 +60,64 @@ const buildCmpBannerConfig = (): CandidateConfig => ({
 	timeoutMillis: null,
 });
 
-const buildReaderRevenueBannerConfig = (
-	CAPI: CAPIBrowserType,
-	isSignedIn: boolean,
-	asyncCountryCode: Promise<string>,
-): CandidateConfig => {
-	return {
-		candidate: {
-			id: 'reader-revenue-banner',
-			canShow: () =>
-				canShowRRBanner({
-					remoteBannerConfig: CAPI.config.remoteBanner,
-					isSignedIn,
-					asyncCountryCode,
-					contentType: CAPI.contentType,
-					sectionName: CAPI.sectionName,
-					shouldHideReaderRevenue: CAPI.shouldHideReaderRevenue,
-					isMinuteArticle: CAPI.pageType.isMinuteArticle,
-					isPaidContent: CAPI.pageType.isPaidContent,
-					isSensitive: CAPI.config.isSensitive,
-					tags: CAPI.tags,
-					contributionsServiceUrl: CAPI.contributionsServiceUrl,
-					alreadyVisitedCount: getAlreadyVisitedCount(),
-					engagementBannerLastClosedAt: getBannerLastClosedAt(
-						'engagementBannerLastClosedAt',
-					),
-					subscriptionBannerLastClosedAt: getBannerLastClosedAt(
-						'subscriptionBannerLastClosedAt',
-					),
-				}),
-			/* eslint-disable-next-line react/jsx-props-no-spreading */
-			show: (meta: any) => () => <ReaderRevenueBanner {...meta} />,
-		},
-		timeoutMillis: DEFAULT_BANNER_TIMEOUT_MILLIS,
+const buildRRBannerConfigWith = ({
+	id,
+	BannerComponent,
+	canShowFn,
+	isEnabled,
+}: RRBannerConfig) => {
+	return (
+		CAPI: CAPIBrowserType,
+		isSignedIn: boolean,
+		asyncCountryCode: Promise<string>,
+	): CandidateConfig => {
+		return {
+			candidate: {
+				id,
+				canShow: () =>
+					canShowFn({
+						remoteBannerConfig: isEnabled(CAPI.config.switches),
+						isSignedIn,
+						asyncCountryCode,
+						contentType: CAPI.contentType,
+						sectionName: CAPI.sectionName,
+						shouldHideReaderRevenue: CAPI.shouldHideReaderRevenue,
+						isMinuteArticle: CAPI.pageType.isMinuteArticle,
+						isPaidContent: CAPI.pageType.isPaidContent,
+						isSensitive: CAPI.config.isSensitive,
+						tags: CAPI.tags,
+						contributionsServiceUrl: CAPI.contributionsServiceUrl,
+						alreadyVisitedCount: getAlreadyVisitedCount(),
+						engagementBannerLastClosedAt: getBannerLastClosedAt(
+							'engagementBannerLastClosedAt',
+						),
+						subscriptionBannerLastClosedAt: getBannerLastClosedAt(
+							'subscriptionBannerLastClosedAt',
+						),
+						section: CAPI.config.section,
+					}),
+				/* eslint-disable-next-line react/jsx-props-no-spreading */
+				show: (meta: any) => () => <BannerComponent {...meta} />,
```

Comment:
isn't `meta` related to `BannerProps`? (which itself contains `meta` prop, which is a little confusing).

If we know what the props are, I think its best to favor directly using the props than spreading them